zio-http/src/main/scala/zhttp/service/Client.scala

```diff
@@ -71,13 +72,25 @@ final case class Client[R](rtm: HttpRuntime[R], cf: JChannelFactory[Channel], el
 
       val isWebSocket = req.url.scheme.exists(_.isWebSocket)
       val isSSL       = req.url.scheme.exists(_.isSecure)
+      val isProxy     = clientConfig.proxyAddress.isDefined
 
       val initializer = new ChannelInitializer[Channel]() {
         override def initChannel(ch: Channel): Unit = {
 
           val pipeline                    = ch.pipeline()
           val sslOption: ClientSSLOptions = clientConfig.ssl.getOrElse(ClientSSLOptions.DefaultSSL)
 
+          // Adding proxy handler
+          if (isProxy)
+            pipeline.addLast(
+              PROXY_HANDLER,
+              new HttpProxyHandler(
+                clientConfig.proxyAddress.getOrElse(
+                  new InetSocketAddress(host, port)
+                )
+              ),
+            )
```

Comment:
Can we also add some tests? You can refer to `ClientSpec`.

Reply:
Hi @tusharmath 

For some reason, `ClientSpec.scala` is just stucking when I'm trying to run it locally. As well as other tests that use `provideLayerShared` method.

I experimented a bit and found that versions before [Updating to ZIO 2.0.0-RC4](https://github.com/dream11/zio-http/pull/1178) works fine

Sorry for disturbing you with that, but maybe you have any CI tool to check if the issue is reproduced only on my local machine? I didn't find it on my own unfortunatelly

Reply:
Hi @tusharmath 

For some reason, `ClientSpec.scala` is just stucking when I'm trying to run it locally. As well as other tests that use `provideLayerShared` method.

I experimented a bit and found that versions before [Updating to ZIO 2.0.0-RC4](https://github.com/dream11/zio-http/pull/1178) works fine

Sorry for disturbing you with that, but maybe you have any CI tool to check if the issue is reproduced only on my local machine? I didn't find it on my own unfortunatelly

Reply:
Added tests to ClientSpec + covered new Proxy class